### PR TITLE
Add overflow option for ModelView toolbar

### PR DESCRIPTION
**Pull request by kisantia**
Add overflow as an option for ModelView toolbar. The default behavior is still wrapping. Overflow can be specified like this:
```
let toolBar = view.modelBuilder.toolbarContainer().withLayout({
    orientation: azdata.Orientation.Horizontal,
    overflow: true
});
```

modelviewToolbarOverflow

### CODE REVIEW

File: src/sql/base/browser/ui/taskbar/media/taskbar.css [hunk 1]
```diff
@@ -129,12 +129,12 @@
...
 }
 
-.carbon-taskbar .overflow .action-item .action-label{
+.carbon-taskbar .toolbarOverflow .action-item .action-label{
 	background-size: 16px;
 	background-position: left;
 }
 
-.overflow .taskbarSeparator {
+.toolbarOverflow .taskbarSeparator {
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/taskbar/media/taskbar.css:137]: Scope this down to .carbon-taskbar

Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/taskbar/media/taskbar.css:137]: Scope this down to .carbon-taskbar

File: src/sql/base/browser/ui/taskbar/overflowActionbar.ts [hunk 2]
```diff
@@ -41,8 +41,8 @@ export class OverflowActionBar extends ActionBar {
 		}));
 
 		this._overflow = document.createElement('ul');
-		this._overflow.id = 'overflow';
-		this._overflow.className = 'overflow';
+		this._overflow.id = 'toolbarOverflow';
+		this._overflow.className = 'toolbarOverflow';
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/taskbar/overflowActionbar.ts:45]: Let's follow standard class naming guidelines and use lowercase, hyphen separated (e.g. `toolbar-overflow`) https://google.github.io/styleguide/htmlcssguide.html#ID_and_Class_Name_Delimiters

File: src/sql/base/browser/ui/taskbar/overflowActionbarStyles.ts [hunk 3]
```diff
@@ -13,28 +13,28 @@ import { focusBorder } from 'vs/platform/theme/common/colorRegistry';
 registerThemingParticipant((theme: IColorTheme, collector: ICssStyleCollector) => {
 	const overflowBackground = theme.getColor(EDITOR_PANE_BACKGROUND);
 	if (overflowBackground) {
-		collector.addRule(`.carbon-taskbar .overflow {
+		collector.addRule(`.toolbarOverflow {
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/taskbar/overflowActionbarStyles.ts:16]: Why did you remove the carbon-taskbar scoping?

File: src/sql/workbench/browser/modelComponents/toolbarContainer.component.ts [hunk 4]
```diff
@@ -47,14 +53,22 @@ export class ToolbarItem {
 				</div>
 			</div>
 			</ng-container>
+			<a #moreActionsButton class="moreActions action-label codicon toggle-more" role="button" tabindex="0" aria-haspopup="true"> </a>
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/modelComponents/toolbarContainer.component.ts:56]: So what we should really be doing here is re-using the actual toolbar component we already define in sql/base/ui. I know that's a good chunk more work but duplicating the code like this just makes it more likely we're going to have issues down the line with fixes/improvements being made in one but the other isn't also fixed/updated.

File: src/sql/base/browser/ui/taskbar/overflowActionbarStyles.ts (hunk 3 above)
Comment (Charles-Gagnon) [reply src/sql/base/browser/ui/taskbar/overflowActionbarStyles.ts:16]: I see - it appears because you want these rules to apply to both the ModelView component and base component? So in general we should really be trying to avoid that - having styles defined here that are used in other completely separate places in the app make it difficult to track dependencies and so it's easy to envision this being changed and breaking the ModelView toolbar but not noticing immediately because it's not clear that this is tied directly to the ModelView toolbar.

If you take my suggestion of re-using the base component in the ModelView component that's less of an issue since you can keep this scoped down and know that the style will apply anywhere that uses this component.

File: src/sql/workbench/browser/modelComponents/toolbarContainer.component.ts (hunk 4 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/browser/modelComponents/toolbarContainer.component.ts:56]: Especially when you see stuff like the duplicated resizing code. That's definitely a huge red flag and should be avoided at all costs.

**Comment by kisantia** [comment]
@Charles-Gagnon I agree that long-term the ModelView toolbar should use the Taskbar for code resuse and maintainability, but that's a much more involved change so I opened a separate task https://github.com/microsoft/azuredatastudio/issues/10109 for that. With other priorities to consider, I think it makes sense to still have this PR to add ModelView toolbar overflow as a stop gap for now so that the toolbar in the dashboard component will support overflow.

File: src/sql/base/browser/ui/taskbar/overflowActionbar.ts (hunk 2 above)
Comment (kisantia) [reply src/sql/base/browser/ui/taskbar/overflowActionbar.ts]: updated name to toolbar-overflow

File: src/sql/base/browser/ui/taskbar/overflowActionbarStyles.ts (hunk 3 above)
Comment (kisantia) [reply src/sql/base/browser/ui/taskbar/overflowActionbarStyles.ts]: Added back the carbon-taskbar scoping and now have the ModelView toolbar theming set separately

**Comment by kisantia** [comment]
Closing because there isn't an immediate need for overflow in ModelView toolbar. Updated the issue #10109 with the two options moving forward:
1. Update the existing ModelView toolbarContainer to use the Taskbar, which would involve needing a way to convert the toolbarContainer items to something the Taskbar accepts(Action or HTMLElement).
2. Create a new ModelView component that is a wrapper around the Taskbar.
